### PR TITLE
fix: Create ConfigMap for GITEA and TEKTON URLs

### LINKED ISSUE #100: core deployment: read the values from configmap
when deploying core service, fuseml-installer reads the yaml file and replaces some strings (e.g. with gitea url). Cleaner solution would be

"creating a config map with those values and making the deployment read it from there." (@flaviodsr )

### PR DESCRIPTION
**Pull request by jsuchome**
No need to parse and sed the yaml template. Closes #100

### CODE REVIEW

File: deployments/core.go [hunk 1]
```diff
@@ -29,6 +27,7 @@ const (
 	coreServiceName         = "fuseml-core"
 	coreServicePort         = 80
 	coreSecretName          = "fuseml-core-gitea"
+	coreConfigMapName       = "config-fuseml-core"
```
Comment (stefannica) [inline deployments/core.go:30]: We already know it's a config, and the name is for a config-map resource type. Does the name really need to reflect that too ? IMO, `fuseml-core` is sufficient.

File: deployments/core.go [hunk 2]
```diff
@@ -242,25 +258,22 @@ func (core Core) apply(c *kubernetes.Cluster, ui *ui.UI, options kubernetes.Inst
...
-	tektonURL, exists := os.LookupEnv("TEKTON_DASHBOARD_URL")
-	if !exists {
-		tektonURL = "http://tekton." + domain
+
+	if err := core.createCoreConfigMap(c, domain); err != nil {
+		return err
 	}
-	if err := core.createCoreDeployment(giteaURL, tektonURL); err != nil {
+
+	//	if err := core.createCoreDeployment(giteaURL, tektonURL); err != nil {
```
Comment (stefannica) [inline deployments/core.go:275]: Leftover debug code ?

**Review by stefannica — COMMENTED** [review]
Great idea, thanks for implementing it. I always felt those regexp replacement bits could use some improvement.

File: deployments/core.go (hunk 1 above)
Comment (jsuchome) [reply deployments/core.go:30]: Hm, I took advice from other configMaps we have, and they ofthen (but not always) have this `config-` prefix... (for example under tekton-pipelines and knative-serving namespaces)

File: deployments/core.go (hunk 2 above)
Comment (jsuchome) [reply deployments/core.go:275]: oops

File: deployments/core.go (hunk 1 above)
Comment (stefannica) [reply deployments/core.go:30]: Ok, I guess it's not such a big deal.

**Review by stefannica — APPROVED** [review]

File: deployments/core.go (hunk 1 above)
Comment (jsuchome) [reply deployments/core.go:30]: I'm just not sure what's the convention. I'm only saying I like it this way little bit more, but I do not _really_ care about it

**Pull request merged**